core/src/telemetry/tracing.ts

```diff
@@ -0,0 +1,389 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * NOTE:
+ *
+ *    We expect that the underlying GenAI SDK will provide a certain
+ *    level of tracing and logging telemetry aligned with Open Telemetry
+ *    Semantic Conventions (such as logging prompts, responses,
+ *    request properties, etc.) and so the information that is recorded by the
+ *    Agent Development Kit should be focused on the higher-level
+ *    constructs of the framework that are not observable by the SDK.
+ */
+
+import {Content} from '@google/genai';
+import {trace, Span, context, Context} from '@opentelemetry/api';
+
+import {BaseAgent} from '../agents/base_agent.js';
+import {InvocationContext} from '../agents/invocation_context.js';
+import {Event} from '../events/event.js';
+import {LlmRequest} from '../models/llm_request.js';
+import {LlmResponse} from '../models/llm_response.js';
+import {BaseTool} from '../tools/base_tool.js';
+import {version} from '../version.js';
+
+// TODO: Replace with constant from opentelemetry.semconv when it reaches version 1.37 in g3.
+const GEN_AI_AGENT_DESCRIPTION = 'gen_ai.agent.description';
+const GEN_AI_AGENT_NAME = 'gen_ai.agent.name';
+const GEN_AI_CONVERSATION_ID = 'gen_ai.conversation.id';
+const GEN_AI_OPERATION_NAME = 'gen_ai.operation.name';
+const GEN_AI_TOOL_CALL_ID = 'gen_ai.tool.call.id';
+const GEN_AI_TOOL_DESCRIPTION = 'gen_ai.tool.description';
+const GEN_AI_TOOL_NAME = 'gen_ai.tool.name';
+const GEN_AI_TOOL_TYPE = 'gen_ai.tool.type';
+
+const tracer = trace.getTracer(
+  'gcp.vertex.agent',
+  version,
+  // TODO: Replace with constant from opentelemetry.semconv when it reaches version 1.37 in g3.
```

Comment:
Do you think this is actual comment?
Looks like a copy-paste artifact that might be removed.